stake-pool/program/tests/vsa_add.rs

```diff
@@ -102,3 +143,411 @@ async fn test_add_validator_stake_account() {
         _ => panic!(),
     }
 }
+
+#[tokio::test]
+async fn test_add_validator_stake_account_with_wrong_token_program_id() {
+    let (
+        mut banks_client,
+        payer,
+        recent_blockhash,
+        stake_pool_accounts,
+        user_stake,
+        user_pool_account,
+    ) = setup().await;
+
+    let mut transaction = Transaction::new_with_payer(
+        &[instruction::add_validator_stake_account(
+            &id(),
+            &stake_pool_accounts.stake_pool.pubkey(),
+            &stake_pool_accounts.owner.pubkey(),
+            &stake_pool_accounts.deposit_authority,
+            &stake_pool_accounts.withdraw_authority,
+            &stake_pool_accounts.validator_stake_list.pubkey(),
+            &user_stake.stake_account,
+            &user_pool_account.pubkey(),
+            &stake_pool_accounts.pool_mint.pubkey(),
+            &stake::id(),
+            &stake::id(),
+        )
+        .unwrap()],
+        Some(&payer.pubkey()),
+    );
+    transaction.sign(&[&payer, &stake_pool_accounts.owner], recent_blockhash);
+    let transaction_error = banks_client
+        .process_transaction(transaction)
+        .await
+        .err()
+        .unwrap();
+
+    match transaction_error {
+        TransportError::TransactionError(TransactionError::InstructionError(
+            _,
+            InstructionError::Custom(error_index),
+        )) => {
+            let program_error = error::StakePoolError::InvalidProgramAddress as u32;
```

Comment:
Why is this `InvalidProgramAddress` and not `IncorrectProgramId`?

Reply:
Same issue with the wrong error from the contract.